addons/account/models/account_move_send.py

```diff
@@ -107,16 +107,31 @@ def _get_alerts(self, moves, moves_data):
         - action the action to run when the link is clicked
         """
         alerts = {}
-        if len(moves) > 1 and (partners_without_mail := moves.filtered(
-                lambda m: 'email' in moves_data[m]['sending_methods'] and not m.partner_id.email).partner_id
-        ):
-            # should only appear in mass invoice sending
-            alerts['account_missing_email'] = {
-                'level': 'warning',
-                'message': _("Partner(s) should have an email address."),
-                'action_text': _("View Partner(s)"),
-                'action': partners_without_mail._get_records_action(name=_("Check Partner(s) Email(s)")),
-            }
+
+        # Filter moves that are trying to send via email
+        email_moves = moves.filtered(lambda m: 'email' in moves_data[m]['sending_methods'])
+        if email_moves:
+            is_batch = len(moves) > 1
+            partners_without_mail = self.env['res.partner']
+
+            # Identify partners without email depending on batch/single send
+            if is_batch:  # Batch sending
```

Comment:
```suggestion
        email_moves = moves.filtered(lambda m: 'email' in moves_data[m]['sending_methods'])
        if email_moves:
            # Identify partners without email depending on batch/single send
            if is_batch := len(moves) > 1:
                # Batch sending
```

`partners_without_mail` gets re-assigned in any case

Reply:
Done :+1: